src/common/snippets/src/lowered/pass/init_loops.cpp

```diff
@@ -38,13 +40,48 @@ int64_t get_output_stride(size_t dim, const VectorDims& shape) {
 
 InitLoops::InitLoops() : Pass() {}
 
-void InitLoops::init_is_incremented(const LinearIR::LoopManager::LoopInfoPtr& loop_info) {
+void InitLoops::init_is_incremented(const LinearIR::LoopManager::LoopInfoPtr& loop_info, size_t loop_id) {
     auto loop_entries = loop_info->get_entry_points();
     auto loop_exits = loop_info->get_exit_points();
-    auto update = [](std::vector<LoopPort>& ports) {
+    auto update = [=](std::vector<LoopPort>& ports) {
         for (auto& port : ports) {
-            if (!ov::is_type<op::MemoryAccess>(port.expr_port->get_expr()->get_node())) {
+            const auto& expr = port.expr_port->get_expr();
+            const auto& expr_loops = expr->get_loop_ids();
+            if (!std::dynamic_pointer_cast<modifier::MemoryAccess>((expr->get_node()))) {
                 port.is_incremented = false;
+            } else if (expr_loops.back() != loop_id) {
+                // Note: LoopPort connected to Buffer between two loops should not be incremented in the outermost loop
```

Comment:
Maybe I missed something but honestly for me it's impossible situation and seems like there is a bug in another place..
Let me explain why:

`LoopPort` is - "plug" (read 'pimpochka') of a Loop that connects to another Loop.
`Store` can store data only to the one memory place and `Load` can read data only from one memory place.
Could you tell me please how `Store` (`Load`) can be `LoopPort` of Loop and store (`load) data to (from) buffer from the same Loop? I really don't understand how  it's possible!

Probably it's special case in TPP but I'm not sure that it's possible!
Could you please explain you case and surprise me? 😃  Maybe offline discussion would be better

Reply:
> Could you tell me please how Store (Load) can be LoopPort of Loop and store (`load) data to (from) buffer from the same Loop? I really don't understand how it's possible!
Easily! Have a look at any MHA pattern with M blocking. You will see that the Buffer on the first Brgemm's output is in the same outermost loop as the softmax's Load.
Of course, I encountered this behavior on TPP, but the comment was simplified to the Load + Store example. Though I wouldn't say that the problem is specific to TPP. It's just that TPP introduced additional complexity to the IR, since now we have 2d and 1D MemoryAccess, and our automatic loop injection can produce really complex multi-level loops.